### ISSUE TITLE
Joining Organisations

### ISSUE THREAD

**Issue by SourSpoon** [padloc/padloc#378]
Hi there,

Issue:
Confirmation Dialogue box never appears when accepting an invite to join a team.

I believe that this is related to using a self-signed certificate. 

I am not exposing the site to the wider web.

I am able to sign-in, create vault items, create teams/ orgs but unable to accept the invite for teams.

```
Time: 2022-01-11T15:04:26.058Z 
Error Code: unknown_error: 
Error Message: Failed to register a ServiceWorker for scope ('https://padloc.holbrookacademy.org/') with script ('https://padloc.holbrookacademy.org/sw.js'): An SSL certificate error occurred when fetching the script.

Time: 2022-01-11T16:33:28.543Z 
Error Code: unknown_error: 
Error Message: Failed to register a ServiceWorker for scope ('https://padloc.holbrookacademy.org/') with script ('https://padloc.holbrookacademy.org/sw.js'): An SSL certificate error occurred when fetching the script.
Stack Trace: 
undefined 

```

**Comment by MaKleSoft** [comment]
Hm, self-signed certificates often do cause problems in unforeseen ways, but I'm not sure why it would affect only that specific functionality. Are the service worker related errors are the only ones you're seeing? Have you tried generating a "proper" certificate using let's encrypt to see if that fixes it?

**Comment by SourSpoon** [comment]
Hi, 
I have dug out our wildcard certificate, but I am still getting similar symptoms,  but the error wont render/ open my mail app when I click on the flag.

I have tried multiple browsers (Chrome v97, Firefox 95.0.2) & multiple PCs.

Are there additional logs I can see?

**Comment by MaKleSoft** [comment]
Can you post the full output from the networks tab and console please?

**Comment by SourSpoon** [comment]
Console:
```
route.ts:73 Uncaught (in promise) DOMException: Failed to execute 'replaceState' on 'History': A history state object with URL 'https://invite/41bcee2c-66fc-4abb-bde6-dc67cbdfac55/0ab07704-2d26-4c0d-a231-bc5e8ca39324' cannot be created in a document with origin 'https://padloc.holbrookacademy.org' and URL 'https://padloc.holbrookacademy.org/unlock?next=%2Finvite%2F41bcee2c-66fc-4abb-bde6-dc67cbdfac55%2F0ab07704-2d26-4c0d-a231-bc5e8ca39324&nobio=1&email=email_redacted%40holbrookacademy.org'.
    at Router.go (https://padloc.holbrookacademy.org/0.chunk.js?__WB_REVISION__=4387dd344d08cf177341429703afbb28:17927:25)
    at https://padloc.holbrookacademy.org/0.chunk.js?__WB_REVISION__=4387dd344d08cf177341429703afbb28:517:65

```

Chrome's exported Har file (email address edited throughout logs)
[padloc.holbrookacademy.org.har.txt](https://github.com/padloc/padloc/files/7854936/padloc.holbrookacademy.org.har.txt)

**Comment by MaKleSoft** [comment]
Ah, there we have it. Looks like you need to set `PL_PWA_URL` on the server side. In your case it would be:
```sh
PL_PWA_URL=https://padloc.holbrookacademy.org
```

**Comment by SourSpoon** [comment]
It is set in my .env does it need to be set elsewhere?
```
# GENERAL SETTINGS
# ================
# URL that will resolve to the server instance
PL_PWA_URL=https://padloc.holbrookacademy.org
# URL that will resolve to the web app
PL_SERVER_URL=https://padloc.holbrookacademy.org/server

# SERVER SETTINGS
# ===============
# Port the server instance will listen on
PL_SERVER_PORT=3000
# Directory where database files will be stored
PL_DB_DIR=./db
# Directory where attachment files will be stored
PL_ATTACHMENTS_DIR=./attachments

# PWA SETTINGS
# ============
# Port the web app will be served from
PL_PWA_PORT=8080
# Directory where the static code for the web app will be stored
PL_PWA_DIR=./pwa

# EMAIL SETTINGS
# =============
# SMTP host
PL_EMAIL_SERVER=
# SMTP username
PL_EMAIL_USER=padloc@holbrookacademy.org
# SMTP port
PL_EMAIL_PORT=587
# SMTP password
PL_EMAIL_PASSWORD=
# Always use TLS when sending emails
PL_EMAIL_SECURE=true

# SSL SETTINGS
# ============
# The path to your ssl certificate
PL_SSL_CERT=/ssl/cert.pem
# The path to your ssl private key
PL_SSL_KEY=/ssl/key.pem
# Advanced configuration file
# PL_SSL_CONF=./ssl/ssl.conf
```

**Comment by SourSpoon** [comment]
It's a holbrookacacademy.org email. Using gmail. It's sending emails OK.

**Comment by MaKleSoft** [comment]
I mean the url that the link/button in the email points to :) Also, the url that is ultimately displayed in your browser, after opening the link (the two may differ).

**Comment by SourSpoon** [comment]
email:
`https://padloc.holbrookacademy.org//invite/41bcee2c-66fc-4abb-bde6-dc67cbdfac55/0ab07704-2d26-4c0d-a231-bc5e8ca39324?email=admin@holbrookacademy.org`

browser (sign-in page)
`https://padloc.holbrookacademy.org/unlock?next=%2Finvite%2F41bcee2c-66fc-4abb-bde6-dc67cbdfac55%2F0ab07704-2d26-4c0d-a231-bc5e8ca39324&nobio=1&email=admin%40holbrookacademy.org`

**Comment by MaKleSoft** [comment]
Hm, looks like there is a extraneous `/` in the original link, so this seems to be related to https://github.com/padloc/padloc/issues/310. What's strange is that in your config `PL_PWA_URL` doesn't have a trailing slash (which could be the cause of this problem). Can you double-check that your `PL_PWA_URL` does **not** have a trailing `/`?

**Comment by SourSpoon** [comment]
I did have a trailing `/` but after your comment two days ago I removed them. Like #310 I followed the guidance on the self hosting page.

I am however still having a range of issues. Which I had hoped to resolve by wiping the box and starting again. 

Symptoms now include: unable to sign up, unable to resend invites to orgs, However I am able to interact with existing vaults and items within them.

I noticed this error in the docker logs.

`2022/01/14 12:19:25 [error] 24#24: *1 connect() failed (111: Connection refused) while connecting to upstream, client: 192.168.60.12, server: _, request: "GET /sw.js HTTP/2.0", upstream: "http://172.18.0.4:8080/sw.js", host: "padloc.holbrookacademy.org", referrer: "https://padloc.holbrookacademy.org/sw.js"`

I hope that's helpful. It appears I must of changed something when troubleshooting
